src/tools/tests/schema2kotlin-test.ts

```diff
@@ -0,0 +1,273 @@
+      `class Handles : HandleHolderBase(
+        "P",
+        mapOf("h1" to P_H1)
+    ) {
+        val h1: ReadSingletonHandle<P_H1> by handles
+    }`
+    ));
+    it('Read, Write and Query Handles', async () => await assertHandleClassDeclaration(
+      `particle P
+        h1: reads Person {name: Text}
+        h2: writes Person {name: Text}
+        h3: reads [Person {name: Text} [name == ?]]
+      `,
+      `class Handles : HandleHolderBase(
+        "P",
+        mapOf("h1" to P_H1, "h2" to P_H2, "h3" to P_H3)
+    ) {
+        val h1: ReadSingletonHandle<P_H1> by handles
+        val h2: WriteSingletonHandle<P_H2> by handles
+        val h3: ReadQueryCollectionHandle<P_H3, String> by handles
+    }`
+    ));
+    it('Handle with references', async () => await assertHandleClassDeclaration(
+      `particle P
+        h1: reads Person {
+          name: Text,
+          home: &Accommodation {
+            squareFootage: Number,
+            address: &Address {
+              streetAddress: Text,
+              postCode: Text  
+            }
+          }
+        }
+      `,
+      `class Handles : HandleHolderBase(
+        "P",
+        mapOf("h1" to P_H1)
+    ) {
+        val h1: ReadSingletonHandle<P_H1> by handles
+    }`
+    ));
+    async function assertHandleClassDeclaration(manifest: string, expectedHandleClass: string) {
+      await assertComponent(manifest, ({handleClassDecl}) => handleClassDecl, expectedHandleClass);
+    }
+  });
+  describe('Schema Aliases', () => {
+    it('Single Entity', async () => await assertSchemaAliases(
+      `particle P
+        h1: reads Person {name: Text}
+      `, [
+        'typealias P_H1 = AbstractP.P_H1'
+      ]
+    ));
+    it('Multiple Connections with the same Schema', async () => await assertSchemaAliases(
+      `particle P
+         h1: reads Person {name: Text}
+         h2: reads [Person {name: Text}]
+         h3: reads [&Person {name: Text}]
+      `, [
+        'typealias P_H1 = AbstractP.PInternal1',
+        'typealias P_H2 = AbstractP.PInternal1',
+        'typealias P_H3 = AbstractP.PInternal1',
+      ]
+    ));
+    it('Handle with references', async () => await assertSchemaAliases(
+      `particle P
+        h1: reads Person {
+          name: Text,
+          home: &Accommodation {
+            squareFootage: Number,
+            address: &Address {
+              streetAddress: Text,
+              postCode: Text  
+            }
+          }
+        }
+      `, [
+        'typealias P_H1 = AbstractP.P_H1',
+        'typealias P_H1_Home = AbstractP.P_H1_Home',
+        'typealias P_H1_Home_Address = AbstractP.P_H1_Home_Address',
+      ]
+    ));
+    async function assertSchemaAliases(manifest: string, expectedAliases: string[]) {
+      await assertComponent(manifest, ({typeAliases}) => typeAliases.sort(), expectedAliases);
+    }
+  });
+  describe('Generated Entity Classes', () => {
+    it('Simple Entity', async () => await assertGeneratedEntity(
```

Comment:
I think this test is of questionable benefit. It strikes me as a change detector test, and because it's so big it's likely we will need to update it a lot. And unlike the golden tests, the failing test won't spit out a handy bash command to update it for you.

Reply:
Yeah, I share the concern. It was even a pain to set the expectation correctly.

The right thing to do may be to break down the entity generation and test separately, the same way we break down the class generation into pieces and test separately. For another day! :) Will remove it now.